addon/PreviewProvider.js

```diff
@@ -58,6 +55,20 @@ PreviewProvider.prototype = {
   },
 
   /**
+    * Builds the current endpoint based on a source
+    */
+  _getCurrentEndpoint(source) {
```

Comment:
Change this to look the source up in a method

getMetadataSource

which just returns the current value of the pref